test/test_id_model.cpp

```diff
@@ -567,36 +630,43 @@ TEST_F(IdModelTest, LoopPromotion1) {
 
   {
     IdModelTester tester(fusion.get());
-    const auto& [iel_graph, root_resolution_map, iel_promotion_map] =
-        tester.getLoopPromotionInfo();
 
     // Nothing inlined. Should be no resolution
-    ASSERT_TRUE(root_resolution_map.empty());
+    ASSERT_TRUE(tester.s1_root_resolution_map.empty());
   }
 
   t2->inlineAt(2);
   ASSERT_EQ(t2->getComputeAtPosition(), 2);
 
   {
     IdModelTester tester(fusion.get());
-    const auto& [iel_graph, root_resolution_map, iel_promotion_map] =
-        tester.getLoopPromotionInfo();
 
     // Check Step 1 results
     // t2 is now fully inlined. Its root broadcast domain should be
     // resoled with the corresponding domain of t3
     validateIELResolution(
         t2->getRootDomain().at(0),
         t3->getRootDomain().at(0),
-        iel_graph,
+        tester.iel_graph,
         tester.idGraph(IdMappingMode::EXACT),
-        root_resolution_map);
+        tester.s1_root_resolution_map);
 
     // Check Step 2 results
     // Nothing to propagate in this fusion, so iel_promotion_map
     // should be equivalent to root_resolution_map
-    ASSERT_EQ(root_resolution_map, iel_promotion_map)
+    ASSERT_EQ(tester.s1_root_resolution_map, tester.s2_iel_promotion_map)
         << "Unexpected IEL promotion map";
+
+    // Check Step 3 results. See the design doc for the expected results
```

Comment:
Instead of saying "see the design doc", should we instead make the nvFuser repo self-contained? Options are:
1. open source the design doc, reformat it as a markdown file in `doc/dev`
2. put the drawings somewhere, for example `doc/tests/xxx.png`, and mention the filename in the comments.

Reply:
Yeah, I'll think about it. I'd wait until all the steps of the loop promotion analysis are completed.